src/poly/dcrt/matrix/generic_matrix.rs

```diff
@@ -303,58 +304,120 @@ impl<T: MatrixElem> GenericMatrix<T> {
         col_start: usize,
         col_end: usize,
     ) -> Self {
-        let nrow = row_end - row_start;
-        let ncol = col_end - col_start;
-        let mut new_matrix = Self::new_empty(&self.params, nrow, ncol);
-        let f = |row_offsets: Range<usize>, col_offsets: Range<usize>| -> Vec<Vec<T>> {
-            let row_offsets = row_start + row_offsets.start..row_start + row_offsets.end;
-            let col_offsets = col_start + col_offsets.start..col_start + col_offsets.end;
+        #[cfg(feature = "disk")]
+        {
+            let nrow = row_end - row_start;
+            let ncol = col_end - col_start;
+            let mut new_matrix = Self::new_empty(&self.params, nrow, ncol);
+            let f = |row_offsets: Range<usize>, col_offsets: Range<usize>| -> Vec<Vec<T>> {
+                let row_offsets = row_start + row_offsets.start..row_start + row_offsets.end;
+                let col_offsets = col_start + col_offsets.start..col_start + col_offsets.end;
 
-            self.block_entries(row_offsets, col_offsets)
-        };
-        new_matrix.replace_entries(0..nrow, 0..ncol, f);
-        new_matrix
+                self.block_entries(row_offsets, col_offsets)
+            };
+            new_matrix.replace_entries(0..nrow, 0..ncol, f);
+            new_matrix
+        }
+        #[cfg(not(feature = "disk"))]
+        {
+            let nrow = row_end - row_start;
+            let ncol = col_end - col_start;
+
+            let mut c = Vec::with_capacity(nrow);
+            for i in row_start..row_end {
+                let mut row = Vec::with_capacity(ncol);
+                for j in col_start..col_end {
+                    row.push(self.inner[i][j].clone());
```

Comment:
Can you use rayon rather than a single-thread for here?
Also, why does this avoid using `replace_entries` when the feature is not disk?